File: Src/ButterflyNet.Parser/Modules/ImageModule.cs
using System;
using System.IO;
using System.Text;

namespace ButterflyNet.Parser.Modules {
	public class ImageModule : IButterflyModule {
		public string Url { get; set; }
		public string Alt { get; set; }
		public string Title { get; set; }
		public int Width { get; set; }
		public int Height { get; set; }

		public void Render(TextWriter writer) {
			if (string.IsNullOrEmpty(Url)) {
				throw new ModuleException("For images, the URL must be specified");
			}

			if (string.IsNullOrEmpty(Alt)) {
				Alt = Url;
			}
			if (string.IsNullOrEmpty(Title)) {
				Title = Url;
			}

			var otherAttributes = new StringBuilder(25);
			if (Width > 0 || Height > 0) {
				otherAttributes.Append(" ");
				if (Width > 0) {
					otherAttributes.Append(string.Format("width=\"{0}\" ", Width));
				}
				if (Height > 0) {
					otherAttributes.Append(string.Format("height=\"{0}\"", Height));
				}
			}

			if (!Url.IsExternalUrl()) {
				Url = "/media/images/" + Url;
			}

			writer.Write(String.Format("<img src=\"{0}\" alt=\"{1}\" title=\"{2}\"{3} />", Url, Alt, Title, otherAttributes.ToString().TrimEnd(' ')));
		}
	}
}

File: src/js/src/modules/ImageModule.js
function ImageModule(baseUrl) {
	ImageModule.$parent.call(this);
	
	this.url = "";
	this.alt = "";
	this.title = "";
	this.width = 0;
	this.height = 0;
	
	this.render = function(writer) {
		if (!this.url) {
			throw new ModuleException("For images, the URL must be specified");
		}
		
		if (!this.alt) {
			this.alt = this.url;
		}
		
		if (!this.title) {
			this.title = this.url;
		}
		
		var otherAttributes = "";
		if (this.width > 0 || this.height > 0) {
			otherAttributes += " ";
			if (this.width > 0) {
				otherAttributes += "width=\"" + this.width + "\" ";
			}
			if (this.height > 0) {
				otherAttributes += "height=\"" + this.height + "\" ";
			}
		}
		
		if (!isExternalUrl(this.url)) {
			this.url = baseUrl + this.url;
		}
		
		writer.write("<img src=\"" + this.url + "\" alt=\"" + this.alt + "\" title=\"" + this.title + "\"" + rtrim(otherAttributes) + " />");
	};
}

extend(Module, ImageModule);